### PR TITLE
🐛  Improve deprecation RegEx

### PR DESCRIPTION
**Pull request by AileenCGN**
closes TryGhost/Ghost#8543

Fixes a bug, where the regex for deprecation checks was wrong because of the usage of `.` in the RegEx rather than `\.` 😔

### CODE REVIEW

**Comment by AileenCGN** [comment]
So, here my two thoughts on this:

This fix _would_ detect the usage of `{{author.image}}` in the exact way, *and only that*. The usage of `{{author_image}}` which is not correct as well (should be `{{author.profile_image}}`), stays then undetected.

If we leave gscan the way it is now, it _would_ detect this false usage, but the message is  confusing, as it explicitly mentions the usage of a detected `{{author.image}}`. 

Would like to here your thoughts about that, @JohnONolan && @kevinansfield !

File: lib/checks/001-deprecations.js [hunk 1]
```diff
@@ -21,32 +21,32 @@ checkDeprecations = function checkDeprecations(theme, themePath) {
                 ruleCode: 'GS001-DEPR-IMG'
             },
             {
-                helperRegEx: '{{author.image}}',
+                helperRegEx: new RegExp(/{{author\.image}}/),
```
Comment (ErisDS) [inline lib/checks/001-deprecations.js]: Escaping the dot will fix the issue :)

Consistency Notes:

- the use of `new RegExp()` isn't really needed, it's useful if you need to build a dynamic regex, e.g. you want to concatenate a variable in like:
`var myRegex = new Regex(myVar1+ myVar1)`
However here I don't think we ever really need it as regexes can be defined as `helperRegex: /{{post\.image}}/`
- Some regexes have the `g` flag, some don't - I think this should be consistent. I believe the `g` flag will make the rule ind ALL instances of the match, and without it will only match the first.
- If you are using `new RegExp()` flags should be passed as the second argument, E.g. `new RegExp(/{{post\.image}}/, 'g')` otherwise it would be `helperRegex: /{{post\.image}}/g`
- `{` is also a special char in regexes, I was going to suggest escaping the first one only to be clear this is not meant to be a special char but testing with https://regex101.com/ shows this is really not necessary.

TLDR; this will fix the bug, there are also some inconsistencies.

**Comment by ErisDS** [comment]
IMO the purpose of the deprecation check is to find themes which are specifically using the old syntax, and tell them about the new syntax. 

Looking for typos or other issues where theme devs have made a mistake is a different problem, which we could look to solve but I think it would be quite hard!

File: lib/checks/001-deprecations.js (hunk 1 above)
Comment (AileenCGN) [inline lib/checks/001-deprecations.js]: Awesome! Thanks a lot! For the sake of consistency, I changed it now, so every regex has the `g` flag.

**Comment by kirrg001** [comment]
@AileenCGN Could you add 1-2 tests after the beta? Thanks 👍

**Comment by AileenCGN** [comment]
Will do!

**Comment by AileenCGN** [comment]
@kirrg001 I added some more test cases in the theme fixtures via #55. Can you check if this is enough?
